optimade/validator/utils.py

```diff
@@ -180,13 +187,15 @@ def __init__(
 
             max_retries: The maximum number of attempts to make for each query.
             headers: Dictionary of additional headers to add to every request.
+            timeout: Connection timeout in seconds.
 
         """
         self.base_url = base_url
         self.last_request = None
         self.response = None
         self.max_retries = max_retries
         self.headers = headers or {}
+        self.timeout = timeout or DEFAULT_CONN_TIMEOUT
```

Comment:
If you do this, just make the init parameter `None` as well, no? Also, while you put it as optional, this does now not allow `None`, since if one passes `timeout=None`, it will autoset it to `DEFAULT_CONN_TIMEOUT` behind the scenes. This is disingenuous I think? Maybe write this in the documentation for the parameter? Hmm, because a maximum timeout would be good to have as a fallback in any case...

Reply:
Indeed, no-one will ever want to put `None` (and mean it) but they may want to put it and not mean it, i.e. use `None` as a blank rather than infinity. Allowing `None` here simplifies the argument pass through at the CLI and in the main validator class. I'd prefer to keep the init value to the actual default that is being used, otherwise the docs suggest that None is the default...